### DISCUSSION TITLE
Enable xdebug in ddev when starting listening in PhpStorm

### DISCUSSION THREAD

**Discussion by dahaupt** [php-perfect/ddev-intellij-plugin#110] [Ideas]
In our projects, we have **Xdebug** disabled by default for performance reasons (as also recommended in the [documentation](https://ddev.readthedocs.io/en/stable/users/debugging-profiling/step-debugging) of **ddev**).

We currently need to manually enable **Xdebug** when starting a debug session in **PhpStorm** with `ddev xdebug on`.

**Proposal:** Enable **Xdebug** in **ddev** automatically when action `Start Listening for PHP Debug Connections` is called in **PhpStorm**.

**Comment by nico-loeber** [comment]
Hey @dahaupt,

With the "right" (my opinion) configuration there is no need to enable/disable xdebug manually, as it can be controlled by cookies/environment varaibles. PHPStorm will already place these cookies/environment vars when clicking on one of the "Debug play buttons".

See 
* https://github.com/drud/ddev/issues/3731

Sorry for my late reply, I didn't receive notifications for these discussions, only for issues, so I missed them. :(

**Comment by dahaupt** [comment]
@nico-loeber Thanks for your response. After reading the mentioned issue, I still think that it is a good practice to disable Xdebug by default for performance reasons. On my system, the request takes clearly longer with Xdebug enabled.

What about adding an option like "Enable Xdebug automatically when starting to listen for debug connections"? 

Feel free to mark this discussion as resolved, in case such a feature is not interesting for the plugin.

**Comment by wernerkrauss** [comment]
IMHO that would be a cool feature. A friend showed that to me in his VSCode setup, and not having to switch on xdebug manually would be pretty comfortable

**Comment by gitressa** [comment]
This would be really nice, since many users probably forget to run `ddev xdebug on` (especially beginners) and then nothing happens ... 

If PhpStorm could take of that step, and only enable Xdebug when needed, i.e. when "Listen for debug connections" is active, it would be an incredible improvement.

**Reply by rfay** [reply]
This does have some downsides. People who don't understand or have a full view of what they're doing get a "hang" if they have a breakpoint set.

**Reply by gitressa** [reply]
Yes, you're right @rfay -- it's not a "black or white" situation, where automating this bit is 100% good ...

How about giving the user a big alert in PhpStorm, if the user activates listening, but Xdebug is not "On" in DDEV?Something like this (if possible)?

The user clicks the "Listen for connections"-icon, and if Xdebug is Off in DDEV: 
> It looks like Xdebug is not enabled in DDEV, this is required for breakpoints to work.
